keras/applications/resnet_rs.py

```diff
@@ -0,0 +1,891 @@
+        bn_axis = 3 if backend.image_data_format() == "channels_last" else 1
+
+        shortcut = inputs
+
+        if use_projection:
+            filters_out = filters * 4
+            if strides == 2:
+                shortcut = layers.AveragePooling2D(
+                    pool_size=(2, 2),
+                    strides=(2, 2),
+                    padding="same",
+                    name=name + "projection_pooling",
+                )(inputs)
+                shortcut = Conv2DFixedPadding(
+                    filters=filters_out,
+                    kernel_size=1,
+                    strides=1,
+                    name=name + "projection_conv",
+                )(shortcut)
+            else:
+                shortcut = Conv2DFixedPadding(
+                    filters=filters_out,
+                    kernel_size=1,
+                    strides=strides,
+                    name=name + "projection_conv",
+                )(inputs)
+
+            shortcut = layers.BatchNormalization(
+                axis=bn_axis,
+                momentum=bn_momentum,
+                epsilon=bn_epsilon,
+                name=name + "projection_batch_norm",
+            )(shortcut)
+
+        # First conv layer:
+        x = Conv2DFixedPadding(
+            filters=filters, kernel_size=1, strides=1, name=name + "conv_1"
+        )(inputs)
+        x = layers.BatchNormalization(
+            axis=bn_axis,
+            momentum=bn_momentum,
+            epsilon=bn_epsilon,
+            name=name + "batch_norm_1",
+        )(x)
+        x = layers.Activation(activation, name=name + "act_1")(x)
+
+        # Second conv layer:
+        x = Conv2DFixedPadding(
+            filters=filters, kernel_size=3, strides=strides, name=name + "conv_2"
+        )(x)
+        x = layers.BatchNormalization(
+            axis=bn_axis,
+            momentum=bn_momentum,
+            epsilon=bn_epsilon,
+            name=name + "batch_norm_2",
+        )(x)
+        x = layers.Activation(activation, name=name + "act_2")(x)
+
+        # Third conv layer:
+        x = Conv2DFixedPadding(
+            filters=filters * 4, kernel_size=1, strides=1, name=name + "conv_3"
+        )(x)
+        x = layers.BatchNormalization(
+            axis=bn_axis,
+            momentum=bn_momentum,
+            epsilon=bn_epsilon,
+            name=name + "batch_norm_3",
+        )(x)
+
+        if 0 < se_ratio < 1:
+            x = SE(filters, se_ratio=se_ratio, name=name)(x)
+
+        # Drop connect
+        if survival_probability:
+            x = layers.Dropout(
+                survival_probability, noise_shape=(None, 1, 1, 1), name=name + "drop"
+            )(x)
+
+        x = layers.Add()([x, shortcut])
+
+        return layers.Activation(activation, name=name + "output_act")(x)
+
+    return apply
+
+
+def BlockGroup(
+    filters,
+    strides,
+    se_ratio,
+    bn_epsilon,
+    bn_momentum,
+    num_repeats,
+    activation,
+    survival_probability: float,
+    name: str,
```

Comment:
Same comment above `name`; most arguments above should have default values

Reply:
Done